autoload/neomake/makers/ft/elixir.vim

```diff
@@ -1,12 +1,34 @@
 " vim: ts=4 sw=4 et
 
+" Credo error types.
+" F -> Refactoring opportunities
+" W -> Warnings
+" C -> Convention violation
+" D -> Software design suggestions
+" R -> Readability suggestions
+" Map structure {CredoError: NeomakeType, ...}
+let s:neomake_elixir_credo_type_map = {
+    \ 'F': 'W',
+    \ 'C': 'W',
+    \ 'D': 'I',
+    \ 'R': 'I'}
+
 function! neomake#makers#ft#elixir#PostprocessEnforceMaxBufferLine(entry) abort
     let buffer_lines = str2nr(line('$'))
     if (buffer_lines < a:entry.lnum)
         let a:entry.lnum = buffer_lines
     endif
 endfunction
 
+function! neomake#makers#ft#elixir#PostprocessCredo(entry) abort
+    let type = toupper(a:entry.type)
+    let type_map = extend(s:neomake_elixir_credo_type_map,
+                \ get(g:, 'neomake_elixir_credo_type_map', {}))
```

Comment:
> Where's the dict there?

I mentioned it's too late to have a dict for configs.  The variable name pattern is well established now.  I was just saying that since we're stuck it should at least be consistent for that reason.

> In the end, if you have something configured in your .credo.exs file, it's because you want to enforce that rule.

Counter argument: I do a lot of Python work in a "project" setting, so I'll have linter configs in the repo dir.  But, I have a lot more one-off scripts that are annoying to see warnings in.  I'd like to be able to pass options to the linter without creating yet another dot file in my home directory.

If it's an option you think people will find useful, it should be there.  Particularly users that aren't inclined to look at Neomake's source to figure things out.